### PR TITLE
Change 9759 to string to build caption

### LINKED ISSUE #9759: Move functional information from toString to getCaption methods II [5]
### Problem Description

Follow-up to #8314: Not all DTOs were covered.

### Proposed Change

- [x] 1. Check and document which classes need to be handled. Refine before to know the effort beforehand.
```
  - DashboardEventDto
  - SystemEventDto - should this be edited ?
  - ContactReferenceDto.PersonName
  - EventIndexDto.EventIndexLocation
  - JournalPersonDto
  - MapContactDto
  - SimilarPersonDto

  - android sync log - should it remain as it was or use the new toString - class + uuid only ?
```

- [x] 2. Adjust `toString` to the usual class + uuid pattern.
- [x] 3. Adjust now not properly formatted values in the vaadin-app (grid columns, selects, maybe labels)
- [x] 4. Adjust now not properly formatted values in the android-app?

### Acceptance Criteria

- [ ] No JPA entity class and no DTO exposes sensitive data in `toString`

### Implementation Details

List of (super-) classes that need to be changed:
1. `ReferenceDto`
2. In the android-app all subclasses of AbstractDomainObject need to be checked (for example `Person`).

Idea: Introduce Interface `HasCaption` for formatting `EntityDto` and `ReferenceDto` classes to ease the use of `CaptionRenderer`.

### PR DESCRIPTION
**Pull request by BarnaBartha**
Fixes #9759

### CODE REVIEW

**Comment by StefanKock** [comment]
I think I won't find the time to review this.

**Comment by MateStrysewske** [comment]
@BarnaBartha Please fix the failing test, I'll start the code review when the CI runs through :)

File: sormas-backend/src/main/java/de/symeda/sormas/backend/docgeneration/TemplateEngine.java [hunk 1]
```diff
@@ -131,6 +131,9 @@ public byte[] generateDocumentDocx(Properties properties, File templateFile) thr
 					} else if (property instanceof String) {
 						property = returnSanitizedString((String) property);
 					}
+//					else if (property instanceof HasCaption) {
```
[the remark this answers is not in the document]
Comment (BarnaBartha) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/docgeneration/TemplateEngine.java]: by mistake, removed now

**Review by MateStrysewske — APPROVED** [review]
Looks good; not merging it ofc because it's supposed to go into 1.79 :)

**Review by syntakker — APPROVED** [review]

**Pull request merged**